src/test/java/fi/otavanopisto/kuntaapi/test/ResourceMocker.java

```diff
@@ -127,33 +138,55 @@ public void setStatus(I id, MockedResourceStatus status) {
       return;
     }
     
-    if (started && resource.getCurrentMapping() != null) {
-      removeStub(resource.getCurrentMapping());
+    if (started && resource.getCurrentMappings() != null) {
+      for (MappingBuilder mapping : resource.getCurrentMappings()) {
+        removeStub(mapping);
+      }
     }
         
     resource.setStatus(status);
     
     if (started) {
-      stubFor(resource.getCurrentMapping());
+      for (MappingBuilder mapping : resource.getCurrentMappings()) {
+        stubFor(mapping);
+      }
+      
       updateStatusLists();
       updateSubMockerStatuses(id, status);
     }
   }
   
   public void mockAlternative(I id, R alternative) {
+    List<MappingBuilder> alternativeMappings = new ArrayList<>();
     MockedResource<R> resource = resources.get(id);
     
-    MappingBuilder okMapping = resource.getMapping(MockedResourceStatus.OK)
-      .willReturn(createOkReturn(alternative));
+    List<MappingBuilder> okMappings = resource.getMappings(MockedResourceStatus.OK);
+    for (MappingBuilder okMapping : okMappings) {
+
+      if (started) {
+        removeStub(okMapping);
+      }
+      
+      MappingBuilder alternativeMapping = null;
+      
+      switch (okMapping.build().getRequest().getMethod().getName()) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1301)